### PR TITLE
ES6 ✨ commercial third party tags

### PR DESCRIPTION
**Pull request by katebee**
## What does this change?

- Converts third-party-tags to ES6
- Goodbye default exports, hello `initThirdPartyTags`
- Renames `init` function from outbrain.js to `initOutbrain` (to add context when it is used)

- ‼️ Introduces a convention for importing the third-party-tag modules, and applies it
(this is why there are so many files changing! Explanation below)
 
Each third-party-tag module will export `url`, `onLoad` and `shouldRun`. We iterate over the modules in the `insertScripts` function, expecting the exports to be named **just so**. However, in third-party-tags we need to avoid the issue of namespace collisions between all these modules.

The problem is that we cannot easily switch to more meaningful names ( e.g. `kruxUrl`, `kruxOnLoad`, `kruxShouldRun`) without re-writing third-party-tags. More background here: https://github.com/guardian/frontend/pull/16996

#### Solution:

The third-party-tag modules now export an **object**, for example:
```
export const imrWorldwide: ThirdPartyTag = {
    shouldRun,
    url,
    onLoad,
};
```

## What is the value of this and can you measure success?

`ThirdPartyTag` is a new Flow type, added to the globals. This approach means Flow takes over the enforcement of the export naming ❤️ `insertScripts` now expects to be passed an `Array<ThirdPartyTag>`, and will complain if the modules deviate from the naming convention 🎉

A target for another PR is to move all the commercial Flow types to their own file, maybe `dev/flow-typed-guardian-frontend/commercial.js`?

- Gets any regressions out of the way ASAP. 😅
- Can convert and expand the tests for third-party-tags

Somewhere for discussion to take place around namespacing conventions.

## Does this affect other platforms - Amp, Apps, etc?

Nope

## Tested in CODE?

- [x] TPT has a lot of responsibility, so test, test, test!

Update 11/07/17: Have tested on CODE, expected scripts appear to be running on the correct pages:

- [x] SimpleReach running on PaidFor content
- [x] IMR and IMR-legacy loading, and sending key values
- [x] Krux is running
- [x] Remarketing is running
- [x] Outbrain is loading

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/third-party-tags.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import * as plista from 'commercial/modules/third-party-tags/plista';
+
+type Service = {
+    shouldRun: boolean,
+    url: string,
+    onLoad?: () => any,
+    useImage?: boolean,
+};
+
+const isLuckyBastard = (): boolean =>
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/third-party-tags.js]: This is still running? 😆

File: static/src/javascripts/projects/commercial/modules/third-party-tags.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        }
+    }
+    if (ref.parentNode) {
+        ref.parentNode.insertBefore(frag, ref);
+    }
+};
+
+const loadOther = (): void => {
+    const services = [
+        audienceSciencePql,
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/third-party-tags.js]: I thought AS line items had been shut down ❔

File: static/src/javascripts/projects/commercial/modules/third-party-tags.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        if (service.useImage) {
+            new Image().src = service.url;
+        } else {
+            const script = document.createElement('script');
+            script.src = service.url;
+            script.onload = service.onLoad;
+            frag.appendChild(script);
+        }
+    }
+    if (ref.parentNode) {
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/third-party-tags.js]: or just `ref.insertAdjacentElement('beforebegin', frag)`

File: static/src/javascripts/projects/commercial/modules/third-party-tags.js (hunk 1 above)
Comment (janua) [inline static/src/javascripts/projects/commercial/modules/third-party-tags.js]: Not at the moment, but it should be this quarter at some point!

**Comment by sndrs** [comment]
i'm not sure about using `*` here, as it means they can never be tree-shook. 

if these modules should export a common shape, is @JonNorman's [suggestion](https://github.com/guardian/frontend/pull/16996#issuecomment-306165306) to add an interface for them the appropriate call?

i think it would give you more safety too, in that what you're looping over would implement what you expect?

**Comment by philwills** [comment]
My instinct was to use a type rather than an interface, as I don't think we'd want to extend this from a class and as far as I can tell that's the main driver for using an interface over a type.

File: static/src/javascripts/projects/commercial/modules/third-party-tags.js (hunk 2 above)
Comment (katebee) [inline static/src/javascripts/projects/commercial/modules/third-party-tags.js]: Modules are still there for now 🙃 

May be a later PR to delete it all from frontend

**Comment by sndrs** [comment]
> Is there some other reason to prefer an interface?

if an interface is only for a class then agree – soryr i should have read the docs further. type sounds good to me?

**Comment by sndrs** [comment]
i guess as a module? https://flow.org/en/docs/libdefs/creation/#toc-declaring-a-module

File: static/src/javascripts/projects/commercial/modules/third-party-tags.js (hunk 3 above)
Comment (katebee) [inline static/src/javascripts/projects/commercial/modules/third-party-tags.js]: Hah. Flow does not like that! 

Do you know why `createDocumentFragment` was used when creating `frag`?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/third-party-tags.js]: Cast the hell out of it 😂

**Review by sndrs — APPROVED** [review]
great stuff @katebee

**Pull request merged**